components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/APIDefinitionFromSwagger12.java

```diff
@@ -0,0 +1,418 @@
+                    String pathContent = new String((byte[]) pathResource.getContent());
+                    JSONObject pathJSON = (JSONObject) parser.parse(pathContent);
+                    pathConfig.put("file", pathJSON);
+                }
+            }
+        } catch (RegistryException e) {
+            handleException("Error while retrieving Swagger Definition for " + apiIdentifier.getApiName() + "-" +
+                    apiIdentifier.getVersion(), e);
+        } catch (ParseException e) {
+            handleException("Error while parsing Swagger Definition for " + apiIdentifier.getApiName() + "-" +
+                    apiIdentifier.getVersion() + " in " + resourcePath, e);
+        }
+        return apiJSON.toJSONString();
+
+    }
+
+    @Override
+    @Deprecated
+    public String createAPIDefinition(API api) throws APIManagementException {
+        JSONParser parser = new JSONParser();
+        String pathJsonTemplate = "{\n    \"path\": \"\",\n    \"operations\": []\n}";
+        String operationJsonTemplate = "{\n    \"method\": \"\",\n    \"parameters\": []\n}";
+        String apiJsonTemplate = "{\n    \"apiVersion\": \"\",\n    \"swaggerVersion\": \"1.2\",\n    \"apis\": [],\n    \"info\": {\n        \"title\": \"\",\n        \"description\": \"\",\n        \"termsOfServiceUrl\": \"\",\n        \"contact\": \"\",\n        \"license\": \"\",\n        \"licenseUrl\": \"\"\n    },\n    \"authorizations\": {\n        \"oauth2\": {\n            \"type\": \"oauth2\",\n            \"scopes\": []\n        }\n    }\n}";
+        String apiResourceJsontemplate = "{\n    \"apiVersion\": \"\",\n    \"swaggerVersion\": \"1.2\",\n    \"resourcePath\":\"\",\n    \"apis\": [],\n    \"info\": {\n        \"title\": \"\",\n        \"description\": \"\",\n        \"termsOfServiceUrl\": \"\",\n        \"contact\": \"\",\n        \"license\": \"\",\n        \"licenseUrl\": \"\"\n    },\n    \"authorizations\": {\n        \"oauth2\": {\n            \"type\": \"oauth2\",\n            \"scopes\": []\n        }\n    }\n}";
+
+
+        APIIdentifier identifier = api.getId();
+
+        APIManagerConfiguration config = ServiceReferenceHolder.getInstance().
+                getAPIManagerConfigurationService().getAPIManagerConfiguration();
+
+        Environment environment = (Environment) config.getApiGatewayEnvironments().values().toArray()[0];
+        String endpoints = environment.getApiGatewayEndpoint();
+        String[] endpointsSet = endpoints.split(",");
+        String apiContext = api.getContext();
+        String version = identifier.getVersion();
+        Set<URITemplate> uriTemplates = api.getUriTemplates();
+        String description = api.getDescription();
+
+        if (endpointsSet.length < 1) {
+            throw new APIManagementException("Error in creating JSON representation of the API" + identifier.getApiName());
+        }
+        if (description == null || description.equals("")) {
+            description = "";
```

Comment:
If an empty value is an accepted option, why do we check if the string is empty in the if clause?